utils/build-script-impl

```diff
@@ -2336,11 +2341,13 @@ for host in "${ALL_HOSTS[@]}"; do
                   continue
                 ;;
                 *)
-                  # FIXME: Always re-build XCTest on non-darwin platforms.
-                  # The Swift project might have been changed, but CMake might
-                  # not be aware and will not rebuild.
-                  echo "Cleaning the XCTest build directory"
-                  call rm -rf "${XCTEST_BUILD_DIR}"
+                  if [[ "${SKIP_CLEAN_XCTEST}" == "0" ]]
+                  then
+                    # The Swift project might have been changed, but CMake might
+                    # not be aware and will not rebuild.
+                    echo "Cleaning the XCTest build directory"
+                    call rm -rf "${XCTEST_BUILD_DIR}"
```

Comment:
This is currently only removed for non-Darwin, mistake? I see llbuild is removed by default for all OSs, I can move this above so it's removed for Darwin too, then this new flag will keep it for both.

Reply:
Moved this up so that this build directory is cleaned by default on macOS too, which is now tested by the updated test.